addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiConnection.java

```diff
@@ -568,57 +568,69 @@ public synchronized void playURI(String uri) {
         socket.callMethod("Player.Open", params);
     }
 
-    private synchronized JsonArray getChannelGroups(final String channelType) {
-        JsonObject params = new JsonObject();
-        params.addProperty("channeltype", channelType);
-        JsonElement response = socket.callMethod("PVR.GetChannelGroups", params);
+    public synchronized List<KodiPVRChannelGroup> getChannelGroups(final String channelType) {
+        String method = "PVR.GetChannelGroups";
+        String hash = method + "#channeltype=" + channelType;
+        JsonElement response = REQUEST_CACHE.putIfAbsentAndGet(hash, () -> {
+            JsonObject params = new JsonObject();
+            params.addProperty("channeltype", channelType);
+            return socket.callMethod(method, params);
+        });
 
+        List<KodiPVRChannelGroup> channelGroups = new ArrayList<>();
         if (response instanceof JsonObject) {
             JsonObject result = response.getAsJsonObject();
-            if (result.has("channelgroups")) {
-                return result.get("channelgroups").getAsJsonArray();
+            for (JsonElement element : result.get("channelgroups").getAsJsonArray()) {
```

Comment:
The old code checks to see if the member `channelgroups` exists in the response. Not being familiar with the protocol, was that check unnecessary? If it's not there, won't this line generate a NPE?

Reply:
In this case it is not crucial. There is always one default channel group of each channel type.